packages/nodes-base/nodes/Ldap/Ldap.node.ts

```diff
@@ -0,0 +1,504 @@
+import { Attribute, Change, Client } from 'ldapts';
+import { ldapFields } from './LdapDescription';
+import { BINARY_AD_ATTRIBUTES, resolveBinaryAttributes } from './Helpers';
+
+export class Ldap implements INodeType {
+	description: INodeTypeDescription = {
+		displayName: 'Ldap',
+		name: 'ldap',
+		icon: 'file:ldap.svg',
+		group: ['transform'],
+		version: 1,
+		subtitle: '={{$parameter["operation"] + ": " + $parameter["resource"]}}',
+		description: 'Interact with LDAP servers',
+		defaults: {
+			name: 'LDAP',
+		},
+		inputs: ['main'],
+		outputs: ['main'],
+		credentials: [
+			{
+				// eslint-disable-next-line n8n-nodes-base/node-class-description-credentials-name-unsuffixed
+				name: 'ldap',
+				required: true,
+				testedBy: 'ldapConnectionTest',
+			},
+		],
+		properties: [
+			{
+				displayName: 'Operation',
+				name: 'operation',
+				type: 'options',
+				noDataExpression: true,
+				options: [
+					{
+						name: 'Compare',
+						value: 'compare',
+						description: 'Compare an attribute',
+						action: 'Compare an attribute',
+					},
+					{
+						name: 'Create',
+						value: 'create',
+						description: 'Create a new entry',
+						action: 'Create a new entry',
+					},
+					{
+						name: 'Delete',
+						value: 'delete',
+						description: 'Delete an entry',
+						action: 'Delete an entry',
+					},
+					{
+						name: 'Rename',
+						value: 'rename',
+						description: 'Rename the DN of an existing entry',
+						action: 'Rename the DN of an existing entry',
+					},
+					{
+						name: 'Search',
+						value: 'search',
+						description: 'Search LDAP',
+						action: 'Search LDAP',
+					},
+					{
+						name: 'Update',
+						value: 'update',
+						description: 'Update attributes',
+						action: 'Update attributes',
+					},
+				],
+				default: 'search',
+			},
+			{
+				displayName: 'Debug',
+				name: 'nodeDebug',
+				type: 'boolean',
+				isNodeSetting: true,
+				default: false,
+				noDataExpression: true,
+			},
+			...ldapFields,
+		],
+	};
+
+	methods = {
+		credentialTest: {
+			async ldapConnectionTest(
+				this: ICredentialTestFunctions,
+				credential: ICredentialsDecrypted,
+			): Promise<INodeCredentialTestResult> {
+				const credentials = credential.data as ICredentialDataDecryptedObject;
+				try {
+					const protocol = credentials.connectionSecurity === 'tls' ? 'ldaps' : 'ldap';
+					const url = `${protocol}://${credentials.hostname}:${credentials.port}`;
+
+					const ldapOptions: ClientOptions = { url };
+					const tlsOptions: IDataObject = {};
+
+					if (credentials.connectionSecurity !== 'none') {
+						tlsOptions.rejectUnauthorized = credentials.allowUnauthorizedCerts === false;
+						if (credentials.caCertificate) {
+							tlsOptions.ca = [credentials.caCertificate as string];
+						}
+						if (credentials.connectionSecurity !== 'startTls') {
+							ldapOptions.tlsOptions = tlsOptions;
+						}
+					}
+
+					const client = new Client(ldapOptions);
```

Comment:
Multiple places where we have duplicate code creating the ldap client. Would be better if we had one createClient() functon.

Reply:
Change is in, thanks.